test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_test.clj

```diff
@@ -642,20 +642,23 @@
         (Files/delete link)))))
 
 (deftest request-logging-test
-  (testing "request logging occurs when :access-log-config is configured"
-    (with-app-with-config
-      app
-      [jetty9-service
-       hello-webservice]
-      {:webserver {:port 8080
-                   :access-log-config
-                         "./dev-resources/puppetlabs/trapperkeeper/services/webserver/request-logging.xml"}}
-      (http-get "http://localhost:8080/hi_world/")
-      ; Logging is done in a separate thread from Jetty and this test. As a result,
-      ; we have to sleep the thread to avoid a race condition.
-      (Thread/sleep 10)
-      (let [list (TestListAppender/list)]
-        (is (re-find #"\"GET /hi_world/ HTTP/1.1\" 200 8\n" (first list)))))))
+  (try
+    (testing "request logging occurs when :access-log-config is configured"
+      (with-app-with-config
+       app
+       [jetty9-service
+        hello-webservice]
+       {:webserver {:port 8080
+                    :access-log-config
+                    "./dev-resources/puppetlabs/trapperkeeper/services/webserver/request-logging.xml"}}
+       (http-get "http://localhost:8080/hi_world/")
+       ; Logging is done in a separate thread from Jetty and this test. As a result,
+       ; we have to sleep the thread to avoid a race condition.
+       (Thread/sleep 10)
+       (let [list (TestListAppender/list)]
+         (is (re-find #"\"GET /hi_world/ HTTP/1.1\" 200 8\n" (first list))))))
+    (finally
+      (.clear (TestListAppender/list)))))
```

Comment:
The `TestListAppender` uses a static `List` to store logged request info.  This line clears out the list before the test ends so that other tests which use the handler don't end up seeing leftover messages logged from other tests.